docs/source/jit.rst

```diff
@@ -531,13 +531,251 @@ Python-defined Constants
+        def foo(len : int) -> torch.Tensor:
+          rv = torch.zeros(3, 4)
+          for i in range(len):
+            if i < 10:
+                rv = rv - 1.0
+            else:
+                rv = rv + 1.0
+          return rv
+
+        print(foo.graph)
+
+    A ``ScriptModule`` with a single ``forward`` method will have an attribute
+    ``graph``, which you can use to inspect the IR representing the computation.
+    If the ScriptModule has more than one method, you will need to access
+    ``.graph`` on the method itself and not the module. We can inspect the
+    graph of a method named ``foo`` on a ScriptModule by accessing ``.foo.graph``.
+
+    The example script above produces the graph::
+
+        graph(%len : int) {
+          %13 : float = prim::Constant[value=1]()
+          %10 : int = prim::Constant[value=10]()
+          %2 : int = prim::Constant[value=4]()
+          %1 : int = prim::Constant[value=3]()
+          %3 : int[] = prim::ListConstruct(%1, %2)
+          %4 : int = prim::Constant[value=6]()
+          %5 : int = prim::Constant[value=0]()
+          %6 : int[] = prim::Constant[value=[0, -1]]()
+          %rv.1 : Dynamic = aten::zeros(%3, %4, %5, %6)
+          %8 : int = prim::Constant[value=1]()
+          %rv : Dynamic = prim::Loop(%len, %8, %rv.1)
+            block0(%i : int, %12 : Dynamic) {
+              %11 : int = aten::lt(%i, %10)
+              %rv.4 : Dynamic = prim::If(%11)
+                block0() {
+                  %14 : int = prim::Constant[value=1]()
+                  %rv.2 : Dynamic = aten::sub(%12, %13, %14)
+                  -> (%rv.2)
+                }
+                block1() {
+                  %16 : int = prim::Constant[value=1]()
+                  %rv.3 : Dynamic = aten::add(%12, %13, %16)
+                  -> (%rv.3)
+                }
+              %19 : int = prim::Constant[value=1]()
+              -> (%19, %rv.4)
+            }
+          return (%rv);
+        }
+
+    Take the instruction ``%rv.1 : Dynamic = aten::zeros(%3, %4, %5, %6)`` for
+    example. ``%rv.1 : Dynamic`` means we assign the output to a (unique)
+    value named ``rv.1``, and that value is of ``Dynamic`` type, i.e. we do
+    not know its concrete shape. ``aten::zeros`` is the operator (equivalent
+    to ``torch.zeros``) and the input list ``(%3, %4, %5, %6)`` specifies which
+    values in scope should be passed as inputs. The schema for built-in functions
+    like ``aten::zeros`` can be found at `Builtin Functions`_.
+
+    Notice that operators can also have associated ``blocks``, namely the
+    ``prim::Loop`` and ``prim::If`` operators. In the graph print-out, these
+    operators are formatted to reflect their equivalent source code forms
+    to facilitate easy debugging.
+
+    Graphs can be inspected as shown to confirm that the computation described
+    by a ``ScriptModule`` is correct, in both automated and manual fashion, as
+    described below.
+
+
+Tracing Edge Cases
```

Comment:
nit: might be nice to have a sphinx warning here for these edge cases

Reply:
Hmm, I just tried this out and it actually doesn't work that well in this context. The warning box seems like it should be in the section that describes the API, whereas this paragraph flows into the next section about automatic checking